website/addons/box/tests/test_views.py

```diff
@@ -0,0 +1,554 @@
+# -*- coding: utf-8 -*-
+"""Views tests for the Box addon."""
+import unittest
+from nose.tools import *  # noqa (PEP8 asserts)
+import mock
+import httplib
+from datetime import datetime
+
+from framework.auth import Auth
+from website.util import api_url_for, web_url_for
+from urllib3.exceptions import MaxRetryError
+from box.client import BoxClientException
+from tests.base import OsfTestCase, assert_is_redirect
+from tests.factories import AuthUserFactory
+
+from website.addons.box.tests.utils import (
+    BoxAddonTestCase, mock_responses, MockBox, patch_client
+)
+from website.addons.box.model import BoxOAuthSettings
+from website.addons.box.views.hgrid import box_addon_folder
+from website.addons.box.views.config import serialize_settings
+
+mock_client = MockBox()
+
+
+class TestAuthViews(OsfTestCase):
+
+    def setUp(self):
+        super(TestAuthViews, self).setUp()
+        self.user = AuthUserFactory()
+        # Log user in
+        self.app.authenticate(*self.user.auth)
+
+    def test_box_oauth_start(self):
+        url = api_url_for('box_oauth_start_user')
+        res = self.app.get(url)
+        assert_is_redirect(res)
+
+    @mock.patch('website.addons.box.views.auth.box_oauth_finish')
+    @mock.patch('website.addons.box.views.auth.finish_auth')
+    @mock.patch('website.addons.box.views.auth.BoxClient')
+    def test_box_oauth_finish(self, mock_get, mock_finish, mock_oauth):
+        mock_client = mock.MagicMock()
+        mock_client.get_user_info.return_value = {'name': 'Mr. Box', 'id': '1234567890'}
+        mock_get.return_value = mock_client
+        mock_finish.return_value = {
+            'token_type': 'something',
+            'access_token': 'something',
+            'refresh_token': 'something'
+        }
+        mock_oauth.return_value = ('mytoken123', 'myboxid', 'done')
+        url = api_url_for('box_oauth_finish')
+
+        res = self.app.get(url)
+
+        assert_is_redirect(res)
```

Comment:
This test could be made more complete, e.g. testing that the redirect URL is correct, the addon was added to the user, etc. @mfraezz Can you please work on this post-merge?